### PR TITLE
#525 grafana module integration

### PR DESCRIPTION
**Pull request by vivek-devtron**
# Description

Divide CICD into 2 part
CICD and Monitoring (Grafana)
Hide metrics from app details and show the option to install grafana
Hide application metrics check from deployment template and deployment template override
remove See metrics for applications in this cluster option from cluster addition

Fixes # [(issue)](https://github.com/devtron-labs/dashboard/issues/525)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?
- Verified hidden metrics section from app details and shown the option to install grafana in case of no grafana module 
- verified hidden application metrics check from deployment template and deployment template override in case of no grafana module 
- verified Remove See metrics for applications in this cluster option from cluster addition

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/deploymentConfig/DeploymentConfig.tsx [hunk 1]
```diff
@@ -229,7 +233,7 @@ export default function DeploymentConfig({
                 {!openComparison && !showReadme && (
                     <DeploymentConfigFormCTA
                         loading={loading || chartConfigLoading}
-                        showAppMetricsToggle={charts && selectedChart && appMetricsEnvironmentVariableEnabled}
+                        showAppMetricsToggle={charts && selectedChart && appMetricsEnvironmentVariableEnabled && (grafanaModuleStatus?.result?.status === ModuleStatus.INSTALLED || false)}
```
Comment (sohel-rp) [inline src/components/deploymentConfig/DeploymentConfig.tsx:236]: Just keep `grafanaModuleStatus?.result?.status === ModuleStatus.INSTALLED `

File: src/components/cluster/Cluster.tsx [hunk 2]
```diff
@@ -379,13 +382,12 @@ function Cluster({
...
                                 active,
                                 config,
-                                environments,
                                 toggleEditMode,
                                 reload,
                                 prometheus_url,
                                 prometheusAuth,
-                                serverMode,
                                 defaultClusterComponent,
+                                isGrafanaModuleInstalled: grafanaModuleStatus?.result?.status === ModuleStatus.INSTALLED || false
```
Comment (sohel-rp) [inline src/components/cluster/Cluster.tsx:390]: same here

File: src/components/EnvironmentOverride/DeploymentTemplateOverride.tsx [hunk 3]
```diff
@@ -209,6 +212,7 @@ export default function DeploymentTemplateOverride({
                     initialise={initialise}
                     handleAppMetrics={handleAppMetrics}
                     handleDelete={handleDelete}
+                    isGrafanaModuleInstalled= {grafanaModuleStatus?.result?.status === ModuleStatus.INSTALLED || false}
```
Comment (sohel-rp) [inline src/components/EnvironmentOverride/DeploymentTemplateOverride.tsx:215]: same here

File: src/components/deploymentConfig/DeploymentConfig.tsx (hunk 1 above)
Comment (vivek-devtron) [reply src/components/deploymentConfig/DeploymentConfig.tsx]: Done

File: src/components/cluster/Cluster.tsx (hunk 2 above)
Comment (vivek-devtron) [reply src/components/cluster/Cluster.tsx]: Done

File: src/components/EnvironmentOverride/DeploymentTemplateOverride.tsx (hunk 3 above)
Comment (vivek-devtron) [reply src/components/EnvironmentOverride/DeploymentTemplateOverride.tsx]: Done

**Review by sohel-rp — APPROVED** [review]
Looks good 👍

File: src/components/app/details/appDetails/AppMetrics.tsx [hunk 4]
```diff
@@ -190,27 +193,31 @@ export const AppMetrics: React.FC<{ appName: string, environment, podMap: Map<st
 
     useEffect(() => {
         let str: string = getCalendarValue(calendarInputs.startDate, calendarInputs.endDate)
```
Comment (shivani170) [inline src/components/app/details/appDetails/AppMetrics.tsx:195]: Can we better rename str?

Comment (vivek-devtron) [reply src/components/app/details/appDetails/AppMetrics.tsx]: Done

**Review by shivani170 — APPROVED** [review]

**Pull request merged**